### PR TITLE
Decision: OCP minor version tracking

### PR DESCRIPTION
**Pull request by bastjan**

### CODE REVIEW

File: docs/modules/ROOT/pages/explanations/decisions/ocp-minor-version-tracking.adoc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+=== Automation in Jira to set cluster facts or update the tenant git repository
+
+Jira is the main tool for tracking customer communication and the status of the upgrade.
+
+We link the Jira ticket to the cluster and set the Lieutenant cluster fact or update the tenant git repository with the required information.
+
+We don't have much experience with Jira automation or workflow, so we don't know how much effort this would be.
+Additionally Jira isn't in our operational control, so we would need to work with other equally busy teams to get this done.
+
+We're also not if Jira is here to stay or if we will switch to another tool in the nearish future.
```
Comment (haasad) [inline docs/modules/ROOT/pages/explanations/decisions/ocp-minor-version-tracking.adoc:29]:
```suggestion
We're also not sure if Jira is here to stay or if we will switch to another tool in the nearish future.
```

**Review by haasad — APPROVED** [review]
Did you decide against including the "admin acknowledgement" process in this decision? IMO it would make sense to have it mentioned here, i.e. that setting the new channel implicitly triggers the admin ack.

File: docs/modules/ROOT/pages/explanations/decisions/ocp-minor-version-tracking.adoc [hunk 2]
```diff
@@ -0,0 +1,79 @@
+= OpenShift Minor Version Upgrade Tracking
+
+== Problem
+
+Upgrading OpenShift minor versions requires a lot of coordination between our OpenShift team, customers, and other teams like AppCat and AppFlow teams.
+It's difficult to track the status of each cluster.
+We need to weekly check if and when we can switch which cluster to a new version and not forget the required pull requests.
```
Comment (simu) [inline docs/modules/ROOT/pages/explanations/decisions/ocp-minor-version-tracking.adoc:7]:
```suggestion
We need to check weekly if and when we can switch which cluster to a new version and not forget the required pull requests.
```

**Comment by bastjan** [comment]
> Did you decide against including the "admin acknowledgement" process in this decision? IMO it would make sense to have it mentioned here, i.e. that setting the new channel implicitly triggers the admin ack.

Good call, needs to be included for completeness.

File: docs/modules/ROOT/pages/explanations/decisions/ocp-minor-version-tracking.adoc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+* Allowing an overview of the status of each cluster
+* Automating the process of changing the release channel
+
+=== Non-Goals
+
+* Automating customer or team approval for the upgrade
+
+== Proposals
+
+=== Automation in Jira to set cluster facts or update the tenant git repository
```
Comment (simu) [inline docs/modules/ROOT/pages/explanations/decisions/ocp-minor-version-tracking.adoc:20]:
```suggestion
=== Automation in Jira to set cluster facts or update the tenant Git repository
```

File: docs/modules/ROOT/pages/explanations/decisions/ocp-minor-version-tracking.adoc [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+Jira is the main tool for tracking customer communication and the status of the upgrade.
+
+We link the Jira ticket to the cluster and set the Lieutenant cluster fact or update the tenant git repository with the required information.
```
Comment (simu) [inline docs/modules/ROOT/pages/explanations/decisions/ocp-minor-version-tracking.adoc:24]:
```suggestion
We link the Jira ticket to the cluster and set the Lieutenant cluster fact or update the tenant Git repository with the required information.
```

File: docs/modules/ROOT/pages/explanations/decisions/ocp-minor-version-tracking.adoc [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+We don't have much experience with Jira automation or workflow, so we don't know how much effort this would be.
+Additionally Jira isn't in our operational control, so we would need to work with other equally busy teams to get this done.
+
+We're also not sure if Jira is here to stay or if we will switch to another tool in the nearish future.
+
+=== Automation in GitLab with metrics export to Grafana dashboard
+
+We create automation to schedule a pull request merge in the tenant git repository to change the release channel.
```
Comment (simu) [inline docs/modules/ROOT/pages/explanations/decisions/ocp-minor-version-tracking.adoc:33]:
```suggestion
We create automation to schedule a pull request merge in the tenant Git repository to change the release channel.
```

File: docs/modules/ROOT/pages/explanations/decisions/ocp-minor-version-tracking.adoc [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+spec:
+  template:
+    spec:
+      channel: stable-4.14
+      clusterID: XXX
+  patches:
+    - from: "2024-07-12T00:00:00Z"
+      patch:
+        spec:
+          channel: stable-4.15
```
Comment (simu) [inline docs/modules/ROOT/pages/explanations/decisions/ocp-minor-version-tracking.adoc:60]: I assume we'll create a proper design page for the feature in a follow-up. I'm not sure if having `patches` as a list is the best option, but it's probably good enough for this use case.

**Review by simu — APPROVED** [review]
some spelling wording suggestions inline.

File: docs/modules/ROOT/pages/explanations/decisions/ocp-minor-version-tracking.adoc (hunk 6 above)
Comment (bastjan) [reply docs/modules/ROOT/pages/explanations/decisions/ocp-minor-version-tracking.adoc:60]: Yes, I'll reference it as implementation idea but not as the finalized solution.

**Pull request merged**